compiler/rustc_trait_selection/src/solve/fulfill.rs

```diff
@@ -149,6 +149,6 @@ impl<'tcx> TraitEngine<'tcx> for FulfillmentCtxt<'tcx> {
         &mut self,
         _: &InferCtxt<'tcx>,
     ) -> Vec<PredicateObligation<'tcx>> {
-        unimplemented!()
+        std::mem::take(&mut self.obligations)
```

Comment:
Just noting that this won't do much, since these are always going to be root obligations. But yeah, no need to panic here.